Release/include/pplx/threadpool.h

```diff
@@ -156,6 +173,9 @@ class threadpool
     std::vector<pthread_t> m_threads;
     boost::asio::io_service m_service;
     boost::asio::io_service::work m_work;
+
+    std::mutex m_connection_pool_map_mutex;
+    std::map<std::string, std::shared_ptr<web::http::client::details::asio_connection_pool>> m_connection_pool_map;
```

Comment:
I'd like to see one more public function added which clears the entire cache. I'm concerned about applications that act more like scanners or crawlers; these applications would basically have a memory leak for every host they connect to.

While finer-grained control of the cache will eventually be needed, I want to ensure we have a workaround available until we can capture some solid cases where the optimal control design becomes clearer.